src/store/middleware.js

```diff
@@ -0,0 +1,40 @@
+import { project } from '@geops/tree-lib';
+
+import {
+  SET_PROJECTION_MODE,
+  SET_MANUAL_LOCATION,
+  SET_MAP_LOCATION,
+  setProjectionResult,
+} from './actions';
+
+const projectionActionTypes = [
+  SET_PROJECTION_MODE,
+  SET_MANUAL_LOCATION,
+  SET_MAP_LOCATION,
+];
+
+export const projection = store => next => action => {
+  const result = next(action);
+  if (projectionActionTypes.includes(action.type)) {
+    const { manualLocation, mapLocation, projectionMode } = store.getState();
+    const location =
+      projectionMode === 'map'
+        ? { ...manualLocation, ...mapLocation }
+        : { ...mapLocation, ...manualLocation };
+    const targetAltitudinalZone =
+      projectionMode === 'map'
+        ? mapLocation.targetAltitudinalZoneModerate
+        : manualLocation.targetAltitudinalZone;
+    try {
+      console.log(location);
```

Comment:
is this there for a purpose?

Reply:
Yeah, until the middleware is more stable.